### PR TITLE
Auth: Add org to role mappings support to GitHub integration 

### PR DESCRIPTION
**Pull request by mgyongyosi**
**What is this feature?**
This PR adds the org role mapping feature to the GitHub connector.

**Why do we need this feature?**

**Who is this feature for?**

**Which issue(s) does this PR fix?**:

**Special notes for your reviewer:**

This is the GitHub part of https://github.com/grafana/grafana/pull/77683.

Relevant PR which added the same org role mapping support Generic OAuth: https://github.com/grafana/grafana/pull/87394.

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md [hunk 1]
```diff
@@ -203,10 +203,9 @@ To map the server administrator role, use the `allow_assign_grafana_admin` confi
...
-This setting denies user access if no role or an invalid role is returned.
+You can disable this default role assignment by setting `role_attribute_strict = true`. This setting denies user access if no role or an invalid role is returned after evaluating the `role_attribute_path` and the `org_mapping` expressions.
 
-You can use the `org_attribute_path` and `org_mapping` configuration options to add roles to other orgs. For more information, refer to [Org roles mapping examples]({{< relref "#org-roles-mapping-examples" >}}). If both org role mapping (`org_mapping`) and the regular role mapping (`role_attribute_path`) are specified, then the user will get the highest of the two mapped roles.
+You can use the `org_attribute_path` and `org_mapping` configuration options to assign the user to organizations and specify their role. For more information, refer to [Org roles mapping example]({{< relref "#org-roles-mapping-example" >}}). If both org role mapping (`org_mapping`) and the regular role mapping (`role_attribute_path`) are specified, then the user will get the highest of the two mapped roles.
```
Comment (jdbaldry) [inline docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md:208]: Unrelated to your change but that use of `relref` is redundant (and discouraged) and could potentially break link resolution.

```suggestion
You can use the `org_attribute_path` and `org_mapping` configuration options to assign the user to organizations and specify their role. For more information, refer to [Org roles mapping example](#org-roles-mapping-example). If both org role mapping (`org_mapping`) and the regular role mapping (`role_attribute_path`) are specified, then the user will get the highest of the two mapped roles.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md [hunk 2]
```diff
@@ -396,10 +391,10 @@ The following table outlines the various generic OAuth2 configuration options. Y
...
+| `org_mapping`                | No       | List of comma- or space-separated `<ExternalOrgName>:<OrgIdOrName>:<Role>` mappings. Value can be `*` meaning "All users". Role is optional and can have the following values: `None`, `Viewer`, `Editor` or `Admin`. For more information on external organization to role mapping, refer to [Org roles mapping example]({{< relref "#org-roles-mapping-example" >}}).                                                                                                                                                                                                                                    |                 |
```
Comment (jdbaldry) [inline docs/sources/setup-grafana/configure-security/configure-authentication/generic-oauth/index.md:397]:
```suggestion
| `org_attribute_path`         | No       | [JMESPath](http://jmespath.org/examples.html) expression to use for Grafana org to role lookup. Grafana will first evaluate the expression using the OAuth2 ID token. If no value is returned, the expression will be evaluated using the user information obtained from the UserInfo endpoint. The result of the evaluation will be mapped to org roles based on `org_mapping`. For more information on org to role mapping, refer to [Org roles mapping example](#org-roles-mapping-example).                                                                                                          |                 |
| `org_mapping`                | No       | List of comma- or space-separated `<ExternalOrgName>:<OrgIdOrName>:<Role>` mappings. Value can be `*` meaning "All users". Role is optional and can have the following values: `None`, `Viewer`, `Editor` or `Admin`. For more information on external organization to role mapping, refer to [Org roles mapping example](#org-roles-mapping-example).                                                                                                                                                                                                                                                   |                 |
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md [hunk 3]
```diff
@@ -114,15 +114,30 @@ To map the server administrator role, use the `allow_assign_grafana_admin` confi
 Refer to [configuration options]({{< relref "#configuration-options" >}}) for more information.
 
 If no valid role is found, the user is assigned the role specified by [the `auto_assign_org_role` option]({{< relref "../../../configure-grafana#auto_assign_org_role" >}}).
-You can disable this default role assignment by setting `role_attribute_strict = true`.
-This setting denies user access if no role or an invalid role is returned.
+You can disable this default role assignment by setting `role_attribute_strict = true`. This setting denies user access if no role or an invalid role is returned after evaluating the `role_attribute_path` and the `org_mapping` expressions.
+
+You can use the `org_attribute_path` and `org_mapping` configuration options to assign the user to organizations and specify their role. For more information, refer to [Org roles mapping example]({{< relref "#org-roles-mapping-example" >}}). If both org role mapping (`org_mapping`) and the regular role mapping (`role_attribute_path`) are specified, then the user will get the highest of the two mapped roles.
```
Comment (jdbaldry) [inline docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md:119]:
```suggestion
You can use the `org_attribute_path` and `org_mapping` configuration options to assign the user to organizations and specify their role. For more information, refer to [Org roles mapping example](#org-roles-mapping-example). If both org role mapping (`org_mapping`) and the regular role mapping (`role_attribute_path`) are specified, then the user will get the highest of the two mapped roles.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md [hunk 4]
```diff
@@ -114,15 +114,30 @@ To map the server administrator role, use the `allow_assign_grafana_admin` confi
...
 
 To ease configuration of a proper JMESPath expression, go to [JMESPath](http://jmespath.org/) to test and evaluate expressions with custom payloads.
 
 #### Role mapping examples
 
 This section includes examples of JMESPath expressions used for role mapping.
 
+##### Org roles mapping example
+
+The GitHub integration does use the external users' teams in the `org_mapping` configuration to map organizations and roles based on their GitHub team membership.
```
Comment (jdbaldry) [inline docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md:129]:
```suggestion
The GitHub integration uses the external users' teams in the `org_mapping` configuration to map organizations and roles based on their GitHub team membership.
```

File: docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md [hunk 5]
```diff
@@ -215,8 +230,9 @@ The table below describes all GitHub OAuth configuration options. Like any other
...
+| `org_mapping`                | No       | List of comma- or space-separated `<ExternalGitHubTeamName>:<OrgIdOrName>:<Role>` mappings. Value can be `*` meaning "All users". Role is optional and can have the following values: `None`, `Viewer`, `Editor` or `Admin`. For more information on external organization to role mapping, refer to [Org roles mapping example]({{< relref "#org-roles-mapping-example" >}}).                                                                                                                                                                                                                                                                                                                                                      |                                               |
```
Comment (jdbaldry) [inline docs/sources/setup-grafana/configure-security/configure-authentication/github/index.md:235]:
```suggestion
| `role_attribute_path`        | No       | [JMESPath](http://jmespath.org/examples.html) expression to use for Grafana role lookup. Grafana will first evaluate the expression using the user information obtained from the UserInfo endpoint. If no role is found, Grafana creates a JSON data with `groups` key that maps to GitHub teams obtained from GitHub's [`/api/user/teams`](https://docs.github.com/en/rest/teams/teams#list-teams-for-the-authenticated-user) endpoint, and evaluates the expression using this data. The result of the evaluation should be a valid Grafana role (`None`, `Viewer`, `Editor`, `Admin` or `GrafanaAdmin`). For more information on user role mapping, refer to [Configure role mapping](#org-roles-mapping-example).                |                                               |
| `role_attribute_strict`      | No       | et to `true` to deny user login if the Grafana org role cannot be extracted using `role_attribute_path` or `org_mapping`. For more information on user role mapping, refer to [Configure role mapping](#org-roles-mapping-example).                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          | `false`                                       |
| `org_mapping`                | No       | List of comma- or space-separated `<ExternalGitHubTeamName>:<OrgIdOrName>:<Role>` mappings. Value can be `*` meaning "All users". Role is optional and can have the following values: `None`, `Viewer`, `Editor` or `Admin`. For more information on external organization to role mapping, refer to [Org roles mapping example](#org-roles-mapping-example).                                                                                                                                                                                                                                                                                                                                                                     |                                               |
```

**Review by jdbaldry — APPROVED** [review]
LGTM

**Review by dmihai — APPROVED** [review]

**Pull request merged**